File: EDA/semana_1/Pruebas/java src/HeapSort.java
public class HeapSort {
    void swap(int arr[], int a, int b){
        int tmp = arr[a];
        arr[a]=arr[b];
        arr[b]=tmp;
    }

    void heapify(int arr[], int n, int i){
        int largo=i;
        int l=2*i+1;
        int r=2*i+2;
        if(l<n && arr[l]>arr[largo]){
            largo=l;
        }
        if(r<n && arr[r]>arr[largo]){
            largo=r;
        }
        if(largo != i){
            swap(arr,i,largo);
            heapify(arr,n,largo);
        }
    }
    public void heapsort(int arr[],int n){
        for(int i=n/2-1;i>=0;i--){
            heapify(arr,n,i);
        }
        for(int i=n-1;i>0;i--){
            swap(arr,0,i);
            heapify(arr,i,0);
        }
    }


    public static void main(String[] args) {
        int arr[]={23,45,7,3,56,98,56,7};
        int n=arr.length;
        HeapSort heap = new HeapSort();
        heap.heapsort(arr,n);
        for(int i=0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
            System.out.println();
    }
}

File: EDA/semana_1/Pruebas/python src/heapSort.py
from timer import Timer

def heapify(arr, n, i):
    largo = i
    l = (2*i)+1
    r = (2*i)+2
    if l < n and arr[l] > arr[largo]:
        largo = l
    if r < n and arr[r] > arr[largo]:
        largo = r
    if largo != i:
        arr[i], arr[largo] = arr[largo], arr[i]
        heapify(arr, n, largo)


def heapsort(arr, n):
    for i in range(n//2+1, -1, -1):
        heapify(arr, n, i)
    for j in range(n-1, 0, -1):
        arr[0], arr[j] = arr[j], arr[0]
        heapify(arr, j, 0)

if __name__ == "__main__":
       
    index = 0
    numArrays = int(input())
    for i in range(numArrays):
        promedio = 0
        arr = []    
        n = int(input())
        for j in range(10):
            aux = input().split(" ")
            for k in range(n):
                arr.append(int(aux[k]))

            t=Timer(n)   
            t.start()
            heapsort(arr, n)
            t.stop()
            promedio += t.printTime()
        print(n,round(promedio/10,3))
        
       


